### ISSUE TITLE
Ruby 3.1 compatibility

### ISSUE THREAD

**Issue by sphynx79** [ElMassimo/better_settings#2]
The new ruby 3.1 use as default gem psych  4.0.3, with this version there is problem to load alias, i receive this error message
Unknown alias: defaults
To fix the problem you can change your  yaml_to_hash method in this way:
```ruby
    # Internal: Parses a yml file that can optionally use ERB templating.
    def yaml_to_hash(file_name)
      return {} if (content = File.open(file_name).read).empty?
      if Psych::VERSION > '4.0'
        YAML.load(ERB.new(content).result, aliases: true).to_hash
      else
        YAML.load(ERB.new(content).result).to_hash
      end
    end
```
In this way it works with ruby 3.1 and keep backwards compatibility

**Comment by ElMassimo** [comment]
Hi Sphynx! @sphynx79

Thanks for reporting and providing a fix.

Happy to accept a PR if you would like to create one, or I can make one with your fix 😃

**Comment by sphynx79** [comment]
Sorry i'am not able to create PR :(

Can you fix the code?

Thanks

**Comment by ElMassimo** [comment]
@sphynx79 Released a fix under `better_settings-1.0.2`, which now requires Ruby 3.1 or greater.
